com.unity.render-pipelines.core/Editor/ShaderStripping/ShaderStrippingReport.cs

```diff
@@ -0,0 +1,333 @@
+                }
+                case ShaderVariantLogLevel.OnlySRPShaders:
+                {
+                    variantsToLog = m_VariantsByPipeline.
+                        Where(i => !string.IsNullOrEmpty(i.Key))
+                        .SelectMany(i => i.Value);
+                    break;
+                }
+            }
+
+            if (variantsToLog != null && variantsToLog.Any())
+            {
+                StringBuilder sb = new StringBuilder();
+                sb.AppendLine($"{this}");
+                foreach (var info in variantsToLog)
+                {
+                    sb.AppendLine($"- {variant}");
+                }
+
+                Debug.Log(sb.ToString());
+            }
+        }
+
+        #region Serialization & Export
+
+        [Serializable]
+        class PipelineVariants
+        {
+            public string pipeline;
+            public ShaderVariantInfo[] variants;
+        }
+
+        [SerializeField] private PipelineVariants[] pipelines;
+        public void OnBeforeSerialize()
+        {
+            pipelines = m_VariantsByPipeline
+                .Select(pipeline => new PipelineVariants() { pipeline = pipeline.Key, variants = pipeline.Value.ToArray() })
+                .ToArray();
+        }
+
+        public void OnAfterDeserialize()
+        {
+            pipelines = null;
+        }
+
+        #endregion
+    }
+
+    /// <summary>
+    /// This class works as an scope of the <see cref="ShaderStrippingReport"/> hooking into the
+    /// <see cref="IPreprocessBuildWithReport"/> that are being called at the begin of the build and
+    /// to <see cref="IPostprocessBuildWithReport"/> that are the ones called after the build is finished
+    /// </summary>
+    class ShaderStrippingReportScope : IPostprocessBuildWithReport, IPreprocessBuildWithReport
+    {
+        internal static bool s_DefaultExport = false;
+        /// <summary>
+        /// Callback order
+        /// </summary>
+        public int callbackOrder => 0;
+
+        /// <summary>
+        /// Creates the Report
+        /// </summary>
+        /// <param name="report">Unused <see cref="BuildReport"/></param>
+        public void OnPreprocessBuild(BuildReport report)
+        {
+            ShaderStrippingReport.instance = new();
+        }
+
+        /// <summary>
+        /// Dumps the report into the console and export the variants
+        /// </summary>
+        /// <param name="report">Unused <see cref="BuildReport"/></param>
+        public void OnPostprocessBuild(BuildReport report)
+        {
+            ShaderVariantLogLevel logStrippedVariants = ShaderVariantLogLevel.Disabled;
+            bool exportStrippedVariants = s_DefaultExport;
+
+            // Check the current pipeline and check the shader variant settings
+            if (RenderPipelineManager.currentPipeline != null && RenderPipelineManager.currentPipeline.defaultSettings is IShaderVariantSettings shaderVariantSettings)
```

Comment:
Potentially you could avoid tracking the data if you checked this in OnPreprocessBuild, would that give any speedups? Now looks like it may be that that all tracking is disabled and you still do the work, only to effectively discard it. Maybe it doesn't make a difference but thought I'd ask.

Reply:
Yeah, I thought that as well, having an empty tracking if both were disabled.